### PR TITLE
Recsys Metric : MRR

### LINKED ISSUE #2631: Implementation of RecSys Metrics
## 🚀 Feature

To add some more metrics used in Recommendation Systems with tests that can directly be used. Here is the list (@alykhantejani feel free to add more):
[] NGCD

### PR DESCRIPTION
**Pull request by shivance**
Fixes #2631 

Adding Mean Reciprocal Rank metric used in Recommendation Systems with tests that can directly be used
[Reference](https://github.com/zuoxingdong/recsys_metrics/blob/master/recsys_metrics/mean_reciprocal_rank.py)

Check list:

- [ ] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
@shivance have you tried to run this locally ? I also wonder about expected result (ref implementation). The link on https://github.com/zuoxingdong/recsys_metrics/blob/master/recsys_metrics/mean_reciprocal_rank.py is not that satisfatory to me. Can you make a quick research on MRR and find another package computing it (something like sklearn)

File: ignite/metrics/mean_reciprocal_rank.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        self._relevance = torch.empty(0)
+
+    @reinit__is_reduced
+    def update(self, output: Sequence[torch.Tensor]) -> None:
+        y_pred, y = output[0].detach(), output[1].detach()
+        _, topk_idx = y_pred.topk(self._k, dim=-1)
+        relevance = y.take_along_dim(topk_idx, dim=-1)
+        self._relevance = torch.cat([self._relevance, relevance], dim=-1)
+        
+    @sync_all_reduce("_sum", "_num_examples")
```
Comment (vfdev-5) [inline ignite/metrics/mean_reciprocal_rank.py:79]: This stuff is incorrect, but let's fix it later once we have a good understanding of how metric is computed
